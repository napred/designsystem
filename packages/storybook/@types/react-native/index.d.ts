// hack because react-native conflicts with @types/node and lib.dom
declare module 'react-native' {
  import { ComponentType } from 'react';

  type Style = {
    default: any;
    create(obj: any): any;
  };

  const StyleSheet: Style;
  const Image: ComponentType;
  const Text: ComponentType;
  const View: ComponentType;

  export { Image, StyleSheet, Text, View };

  export type RegisteredStyle<T> = any;
}
