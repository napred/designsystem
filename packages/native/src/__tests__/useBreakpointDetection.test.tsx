import React from 'react';
import { Dimensions, Text } from 'react-native';
import { render } from 'react-native-testing-library';
import { createComponent, DesignSystem } from '../';

jest.mock('Dimensions');

const DSText = createComponent('DSText', Text);

describe('useBreakpointDetection', () => {
  it('works correctly', () => {
    Dimensions.set({
      screen: {
        fontScale: 1,
        height: 667,
        scale: 3,
        width: 375,
      },
      window: {
        fontScale: 1,
        height: 667,
        scale: 3,
        width: 375,
      },
    });
    const tree = (
      <DesignSystem>
        <DSText color={['tomato', 'orchid', 'navy', 'cornsilk']} />
      </DesignSystem>
    );
    const { toJSON, unmount, update } = render(tree);

    expect(toJSON()).toMatchSnapshot();

    update(tree); // component did mount

    // change to ipad
    Dimensions.set({
      screen: {
        fontScale: 1,
        height: 1366,
        scale: 2,
        width: 1024,
      },
      window: {
        fontScale: 1,
        height: 1366,
        scale: 2,
        width: 1024,
      },
    });
    // @ts-ignore
    Dimensions._update();

    expect(toJSON()).toMatchSnapshot();

    // change to something bigger
    Dimensions.set({
      screen: {
        fontScale: 1,
        height: 1080,
        scale: 1,
        width: 1920,
      },
      window: {
        fontScale: 1,
        height: 1080,
        scale: 1,
        width: 1920,
      },
    });

    // @ts-ignore
    Dimensions._update();

    expect(toJSON()).toMatchSnapshot();

    unmount();
  });
});
