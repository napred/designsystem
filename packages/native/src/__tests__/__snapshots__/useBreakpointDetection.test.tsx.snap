// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useBreakpointDetection works correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "tomato",
        "fontFamily": "Palanquin",
        "fontSize": 16,
        "lineHeight": 24,
      },
    ]
  }
/>
`;

exports[`useBreakpointDetection works correctly 2`] = `
<Text
  style={
    Array [
      Object {
        "color": "orchid",
        "fontFamily": "Palanquin",
        "fontSize": 16,
        "lineHeight": 24,
      },
    ]
  }
/>
`;

exports[`useBreakpointDetection works correctly 3`] = `
<Text
  style={
    Array [
      Object {
        "color": "cornsilk",
        "fontFamily": "Palanquin",
        "fontSize": 16,
        "lineHeight": 24,
      },
    ]
  }
/>
`;
